server/src/main/java/org/elasticsearch/index/mapper/FieldNamesFieldMapper.java

```diff
@@ -155,40 +156,14 @@ public FieldNamesFieldType fieldType() {
         return (FieldNamesFieldType) super.fieldType();
     }
 
-    static Iterable<String> extractFieldNames(final String fullPath) {
-        return new Iterable<String>() {
-            @Override
-            public Iterator<String> iterator() {
-                return new Iterator<>() {
-                    int endIndex = nextEndIndex(0);
-
-                    private int nextEndIndex(int index) {
-                        while (index < fullPath.length() && fullPath.charAt(index) != '.') {
-                            index += 1;
-                        }
-                        return index;
-                    }
-
-                    @Override
-                    public boolean hasNext() {
-                        return endIndex <= fullPath.length();
-                    }
-
-                    @Override
-                    public String next() {
-                        final String result = fullPath.substring(0, endIndex);
-                        endIndex = nextEndIndex(endIndex + 1);
-                        return result;
-                    }
-
-                    @Override
-                    public void remove() {
-                        throw new UnsupportedOperationException();
-                    }
-
-                };
-            }
-        };
```

Comment:
on simplifying this and not indexing all the inner paths, that makes sense, but should we make that change in a separate PR?

Reply:
That's a good point, I'll pull this change into a separate issue.